src/grids/linear_grid.jl

```diff
@@ -1,217 +1,218 @@
+            fft_x⁻¹, ft_arr, data_range)
+    end
 end
 
 """
-    LinearGrid(xmin::Real, xmax::Real, len::Int; complex_op=false, T::Type=Float64)
+    LinearGrid(grid_min::Real, len::Int; complex_op=true, T::Type=Float64)
 
-Generates an equidistant grid from `xmin` to `xmax` with `len` points. If operations on
-complex arrays are expected, `complex_op` should be set to `true`.
+Generates a symmetric equidistant grid from `grid_min` to `-grid_min` with `len` points.
+If operations on real arrays are expected, `complex_op` should be set to `false`.
 
 # Fields
-- `xmin`, `xmax`, `len` defined above.
+- `grid_min`, `len` defined above.
 - `points`: Vector of grid points `pᵢ`.
-- `δ`: Discretisation parameter of `points`, i.e. `pᵢ₊₁ = pᵢ + δ`.
+- `d`: Discretisation parameter of `points`, i.e. `pᵢ₊₁ = pᵢ + d`.
 - `op`: Tools for operations on this `LinearGrid`.
 
 See also [`LinearGridOps`](@ref).
 """
 struct LinearGrid{T} <: SupportGrid{T}
-  xmin::T               # Left bound
-  xmax::T               # Right bound
-  len::Int              # Number of points
-  points::Vector{T}     # Array of grid points
-  δ::T                  # Frequency discretisation
-  op::LinearGridOps{T}
-
-  function LinearGrid(xmin::Real, xmax::Real, len::Int;
-      complex_op::Bool=false, T::Type=Float64
+    grid_min::T            # Left bound
+    len::Int              # Number of points
+    points::Vector{T}     # Array of grid points
+    d::T                  # Discretisation parameter
+    op::LinearGridOps{T} # Tools for FFT operations on this grid
+
+    function LinearGrid(grid_min::Real, len::Int;
+        complex_op::Bool=true, T::Type=Float64
     )
-    indices = 0:(len-1)
-    
-    δ = (xmax - xmin) / (len - 1)
-    points = δ .* collect(indices) .+ xmin
-    points[end] = xmax
-    
-    new{T}(xmin, xmax, len, points, δ, LinearGridOps(points, complex_op; δ))
-  end
-  #
-  LinearGrid(::Type{T}, args...; kwargs...) where T = LinearGrid(args...; T, kwargs...)
+
+        indices = 0:(len-1)
+
+        d = 2abs(grid_min) / len
+
+        # Symmetric grid around 0, note that the last point is not included
+        points = d .* (collect(indices) .- len ÷ 2)
+
+        new{T}(grid_min, len, points, d, LinearGridOps(points, complex_op; d))
+    end
+    #
+    LinearGrid(::Type{T}, args...; kwargs...) where {T} = LinearGrid(args...; T, kwargs...)
 end
 
 """
 Apply FFT of grid to input
 """
-function fft!(grid::LinearGrid, out::AbstractVector, u::AbstractVector)
-  (; padded, ft, ft⁻¹, ft_shift) = grid.op
-  
-  assert_applicable(ft, padded, out)
-  unsafe_execute!(ft, _zeropad!(padded, u), out)
-  @. out *= ft_shift
-  
-  return out
+function ft!(grid::LinearGrid, out::AbstractVector, u::AbstractVector)
+    return fftshift(bfft(u)) .* grid.op.ft_arr[1:length(u)] * grid.d
 end
 
-fft(grid::LinearGrid, u::AbstractVector) = fft!(grid, zero(grid.op.ft_work1), u)
+ft(grid::LinearGrid, u::AbstractVector) = ft!(grid, zero(u), copy(u))
```

Comment:
Why the copy?

Reply:
To be honest, it is not necessary. 
We have to multiply u by ft_arr before we fourier transform, so we either just multiply (not in place) u by ft_arr before fft or like I did here create a copy which is then multiplied in place. 
Maybe there is a smarter way?

Reply:
In this specific case it actually just calls bfft on u, which should be non-altering. But yes, when we use the plan in the future we need to think about this actually.